hcl2template/types.build.post-processor.go

```diff
@@ -4,27 +4,43 @@ import (
 	"fmt"
 
 	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
 	"github.com/hashicorp/packer/packer"
 )
 
 // PostProcessorBlock represents a parsed PostProcessorBlock
 type PostProcessorBlock struct {
 	PType string
-	block *hcl.Block
+	PName string
+
+	HCL2Ref
+}
+
+func (p *PostProcessorBlock) String() string {
+	return fmt.Sprintf(buildPostProcessorLabel+"-block %q %q", p.PType, p.PName)
 }
 
-func (p *Parser) decodePostProcessorGroup(block *hcl.Block) (*PostProcessorBlock, hcl.Diagnostics) {
+func (p *Parser) decodePostProcessor(block *hcl.Block) (*PostProcessorBlock, hcl.Diagnostics) {
+	var b struct {
+		Name string   `hcl:"name,optional"`
+		Rest hcl.Body `hcl:",remain"`
+	}
+	diags := gohcl.DecodeBody(block.Body, nil, &b)
```

Comment:
So this basically allows to partially load an hcl block; allowing to do what we want; I also did that for provisioners since it's the exact same change.

Reply:
oh, interesting! I was trying to add a labelName but it didn't appear we could do that and make it optional. this approach seems good.